bibliopixel/remote/control.py

```diff
@@ -0,0 +1,140 @@
+import multiprocessing
+import inspect
+from .. animation import BaseAnimation
+from .. animation.animation import COMPLETE_REASON
+from . import server
+from .. import log
+
+
+DEFAULT_OFF = 'OFF_ANIM'
+DEFAULT_ANIM_CONFIG = {
+    'bgcolor': '#00ff00',
+    'font_color': '#ffffff',
+    'display': None
+}
+
+
+class RemoteControl():
+    def __init__(self, base_config, animations):
+        # These start with the defaults
+        self.config = {
+            'title': 'BiblioPixel Remote',
+            'bgcolor': '#000000',
+            'font_color': '#ffffff',
+            'external_access': False,
+            'port': 5000
+        }
+
+        self.config.update(base_config)
+
+        self.ui_config = {
+            'bgcolor': self.config['bgcolor'],
+            'title': self.config['title']
+        }
+
+        self.q_send = multiprocessing.Queue()
+        self.q_recv = multiprocessing.Queue()
+
+        server_args = (
+            self.config['external_access'],
+            self.config['port'],
+            self.q_send,
+            self.q_recv
+        )
+
+        self.server = multiprocessing.Process(target=server.run_server,
+                                              args=server_args)
+
+        if not isinstance(animations, list):
+            raise ValueError('animations must be a list!')
+
+        self.current_animation_name = None
```

Comment:
Because the obj is just a BaseAnimation class instance (or inherited version). That doesn't have a name. The name here is the name configured in the remote json config. I didn't want to add an extraneous property called `name` to BaseAnimation since it doesn't care.